File: src/python/inputs-complex/zlib-1.3.1/individual-funcs/gzeof.c
int ZEXPORT gzeof(gzFile file) {
    gz_statep state;

    /* get internal structure and check integrity */
    if (file == NULL)
        return 0;
    state = (gz_statep)file;
    if (state->mode != GZ_READ && state->mode != GZ_WRITE)
        return 0;

    /* return end-of-file state */
    return state->mode == GZ_READ ? state->past : 0;
}

File: src/python/inputs-complex/zlib-1.3.1/individual-funcs/gzeof.rs
fn gzeof(file: *mut GzFile) -> i32 {
    let state = file.as_mut().unwrap();

    if state.mode != GZ_READ && state.mode != GZ_WRITE {
        return 0;
    }

    if state.mode == GZ_READ {
        return state.past;
    } else {
        return 0;
    }
}